### ISSUE TITLE
Bad redirect on login/logout when using a reverse proxy

### ISSUE THREAD

**Issue by jant90** [theotherp/nzbhydra2#214]
I just made the switch from NZBHydra to NZBHydra 2 and I'm impressed with its speed, however I noticed a bug (I think).

I'm running NZBHydra 2 behind a reverse proxy using Apache 2 to add SSL and use my own domain name.

When logging the form's request URL is (while using the 'login form' auth type):
https://mydomain.tld/nzbhydra2/login
(where the login succeeds)

But then I'm redirected to:
http://mydomain.tld:5076/nzbhydra2/
(which is not reachable so the web page shows an unknown error)

I have to manually refresh the page and then I'm logged in.

I see two issues here. First the port is included in the URL and second the URL isn't HTTPS. The same happens when logging out.

When sending an NZB so my download client I initially had the same issue but I think that was solved by adding the following values to my Apache config, unfortunately that didn't fix the login redirect:
```
ProxyPreserveHost On
RequestHeader set X-Forwarded-Proto https
```

This is what my Apache reverse proxy config looks like:
```
<Location /nzbhydra2/>
	ProxyPreserveHost On
	RequestHeader set X-Forwarded-Proto https
	ProxyPass http://localhost:5076/nzbhydra2/
	ProxyPassReverse http://localhost:5076/nzbhydra2/
</Location>
Redirect 301 /nzbhydra2 /nzbhydra2/
Redirect 301 /nzbhydra /nzbhydra2/
```

The NZBHydra log doesn't show anything related to this.

**Comment by theotherp** [comment]
Please follow the instructions here: https://github.com/theotherp/nzbhydra2/wiki/Exposing-Hydra-to-the-internet-and-using-reverse-proxies

**Comment by jant90** [comment]
Thanks, I looked at that and thought I had adopted everything that was required from it. I started with the config I used with NZBHydra 1 where a simple location directive with ProxyPass settings was enough, like so:
```
<Location /nzbhydra/>
	ProxyPass http://localhost:5075/nzbhydra
	ProxyPassReverse http://localhost:5075/nzbhydra
</Location>
```
I added the values I found in the wiki to get it working with NZBHydra 2.

That configuration felt safe because each proxied service has its own location directive where I can add custom/unique settings for each proxied service.

Anyways, after some more trial and error I found that the only issue with my initial config was that I used `localhost` instead of `127.0.0.1` for the ProxyPass values. After changing that it was working fine. And after changing NZBHydra 2's host setting from `0.0.0.0` to `127.0.0.1` I could use `localhost` in the ProxyPass settings in Apache again.

**So I guess the lesson is use `127.0.0.1` for the Apache proxy settings, OR if you want to use `localhost` there first set NZBHydra 2's host to `127.0.0.1`.**
